File: examples/.scripts/timestamp-echo-native/timestamp_echo.ex
defmodule Examples.Skills.TimestampEcho do
  @moduledoc """
  Native Elixir implementation of the timestamp-echo skill.

  Same behavior as the Python version - echoes a message with a timestamp.
  Runs directly in the BEAM with no external processes.
  """

  @behaviour Conjure.NativeSkill

  @impl true
  def __skill_info__ do
    %{
      name: "timestamp-echo",
      description: "Echo a message with the current timestamp",
      allowed_tools: [:execute]
    }
  end

  @impl true
  def execute(message, _context) do
    timestamp = Calendar.strftime(DateTime.utc_now(), "%Y-%m-%d %H:%M:%S")
    {:ok, "[#{timestamp}] #{message}"}
  end
end

File: examples/skills/timestamp-echo/scripts/timestamp_echo.py
#!/usr/bin/env python3
"""
Timestamp Echo Skill - Echoes a message with the current timestamp.

Usage:
    python3 timestamp_echo.py [message]
"""

import sys
from datetime import datetime


def timestamp_echo(message: str = "") -> str:
    """Echo a message with the current timestamp."""
    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    if message:
        return f"[{timestamp}] {message}"
    return f"[{timestamp}]"


def main():
    message = " ".join(sys.argv[1:]) if len(sys.argv) > 1 else ""
    result = timestamp_echo(message)
    print(result)


if __name__ == "__main__":
    main()
